src/config/config.js

```diff
@@ -0,0 +1,74 @@
+const validators = {
+  GATEWAY_BASE_URL: (value) => {
```

Comment:
This should tackle the concerns mentioned in the previous comments:

[#1](https://github.com/pokt-network/leif/pull/99\#discussion_r626851093)
[#2](https://github.com/pokt-network/leif/pull/99\#discussion_r626850205)

I will keep those comments unresolved until you approve those new changes.

Reply:
This seems like a lot of code vs. just changing the GATEWAY BASE URL in each environment?

I'm trying to imagine a situation where you'd change the chain in the same environrment but I can't.

Reply:
Response to this is in the PR's discussion thread. (_noting for trace_)